### DISCUSSION TITLE
Area/Volume fraction

### DISCUSSION THREAD

**Discussion by Emmanuel-324** [idaholab/moose#28582] [Q&A Modules: Phase field] [answered]
Is there a kernel/auxkernel or postprocessor that can help me calculate the volume/area fraction of my precipitates during or after evolution of the precipitates?

**Comment by GiudGiud** [comment, marked as answer]
https://mooseframework.inl.gov/source/postprocessors/AverageGrainVolume.html

**Reply by GiudGiud** [reply]
and https://mooseframework.inl.gov/source/postprocessors/GrainBoundaryArea.html

**Reply by Emmanuel-324** [reply]
Is this in my postprocessor going to calculate the area of my eta(order parameter) as they evolve in the matrix?

```
 [./gr1area]
    type = ElementIntegralVariablePostprocessor
    variable = eta1
    execute_on = 'initial timestep_end'
  [../]
  [./gr2area]
    type = ElementIntegralVariablePostprocessor
    variable = eta2
    execute_on = 'initial timestep_end'
  [../]
  [./gr3area]
    type = ElementIntegralVariablePostprocessor
    variable = eta3
    execute_on = 'initial timestep_end'
  [../] 
``` 

And I assume this also gives the average volume of the precipitates as they grow?
```
[./avg_grain_vol]
    type = AverageGrainVolume
    grain_num = 3
    execute_on = 'initial timestep_end'
  [../]
```

**Reply by GiudGiud** [reply]
Yes I think both of these should be good
